### PR TITLE
MULTIARCH-3789: Power VS: Remove cloud connection support

### PR DESCRIPTION
**Pull request by mjturek**
Removal of remaining references to cloud connections and zones that required them.

Closes https://issues.redhat.com/browse/MULTIARCH-3789

### CODE REVIEW

**Review by r4f4 — COMMENTED** [review]
/approve

File: data/data/powervs/cluster/power_network/pi_network.tf [hunk 1]
```diff
@@ -8,32 +8,15 @@ data "ibm_pi_dhcps" "dhcp_services" {
 }
 
 resource "ibm_pi_dhcp" "new_dhcp_service" {
-  count                  = 1
-  pi_cloud_instance_id   = var.cloud_instance_id
-  pi_cloud_connection_id = var.transit_gateway_enabled ? "" : data.ibm_pi_cloud_connection.cloud_connection[count.index].id
```
Comment (miyamotoh) [inline data/data/powervs/cluster/power_network/pi_network.tf:13]: So, you no longer reference `transit_gateway_enabled` in this module, and thus you should drop the variable from `variables.tf` in this directory and also from the parent `main.tf` calling this, right?

**Review by miyamotoh — APPROVED** [review]
Generally LGTM, except the `transit_gateway_enabled` ref in the `pi_network` module which appears to be no longer needed/used.

File: data/data/powervs/cluster/power_network/pi_network.tf (hunk 1 above)
Comment (mjturek) [reply data/data/powervs/cluster/power_network/pi_network.tf:13]: @miyamotoh hmm you're right but could we do a small follow up patch that removes references to `transit_gateway_enabled`? The patch as is is tested and critical for 4.15 deliverables

**Pull request merged**